### DISCUSSION TITLE
Creating a github page for hosting community trained models

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#304] [Archived]
<USER>>>> erogol<br/> <DATE> [November 20, 2020, 12:40pm]<br/> <br/> I think we can create a GitHub page to host community-driven models<br/> (including models I trained) to enable a better distribution.<br/> <br/> I wonder to see who would like to share models and help on that?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/creating-a-github-page-for-hosting-community-trained-models]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [November 20, 2020, 12:48pm]<br/> <br/> maybe we can incorporate models you trained<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> mrthorstenm<br/> <DATE> [November 20, 2020, 2:28pm]<br/> <br/> . *slash* <br/> I've already put some german models i know (*based on my dataset*) on my<br/> github page here:<br/> <br/> [deep-learning-german-tts](https://thorstenmueller.github.io/deep-learning-german-tts/audio_compare)<br/> <br/> ### [Vocoder Vergleich auf Basis des 'thorsten' Tacotron 2 Modells](https://thorstenmueller.github.io/deep-learning-german-tts/audio_compare)<br/> <br/> The free german voice dataset.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> synesthesiam<br/> <DATE> [November 20, 2020, 3:02pm]<br/> <br/> for all of the<br/> excellent work on MozillaTTS<br/> <br/> Sure, I'd be happy to contribute! ...the only problem is that the models<br/> I've trained are not compatible with upstream MozillaTTS. The vocoders<br/> should be fine, however.<br/> <br/> I made [a few tweaks](https://github.com/rhasspy/TTS) to have more<br/> control over the phonemes. Specifically:<br/> <br/> [gruut](https://github.com/rhasspy/gruut) instead of phonemizer<br/> in the text utils<br/> of EOS/BOS symbols<br/> <br/> Mostly, these changes ensure that the `characters.phonemes` list is<br/> preserved in order, and that nothing (besides the pad symbol) is<br/> automatically added.<br/> <br/> But the use of `gruut` over `phonemizer` is probably going to be a show<br/> stopper for most people. Phonemes in `gruut` come from pre-built<br/> dictionaries or pre-trained grapheme-to-phoneme models, which lets me do<br/> some neat things like apply accents to voices. It also does<br/> tokenization, text cleaning, and number/currency expansion with the help<br/> of Babel and num2words.<br/> <br/> Let me know how I can help<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [November 24, 2020, 1:10am]<br/> <br/> - you suggested<br/> gruut might be a 'show stopper for most people'. How does it compare<br/> with using phonemizer with espeak-ng as a backend?<br/> <br/> Is one of the concerns that it doesn't have such broad language<br/> coverage?<br/> <br/> If the phoneme symbols are consistent then presumably people can switch<br/> back and forth between it and phonemizer to see how it compares - I<br/> would be interested to give that a go, is there anything I should bear<br/> in mind when trying it?<br/> <br/> Happy to move this discussion into a separate thread if that's better.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> synesthesiam<br/> <DATE> [November 24, 2020, 4:02pm]<br/> <br/> > How does it compare with using phonemizer with espeak-ng as a backend?<br/> <br/> Both gruut and phonemizer produce IPA, but gruut uses pre-built lexicons<br/> and g2p models. I haven't tested how consistent the IPA is between the<br/> two, but I'd expect it to be pretty good for U.S. English (gruut's U.S.<br/> English phoneme inventory is<br/> [here](https://github.com/rhasspy/gruut/blob/master/gruut/data/en-us/phonemes.txt)).<br/> <br/> For me, an important feature of gruut is that a word can have multiple<br/> pronunciations. 'read', for example, has both `/ɹɛd/` (like 'red') and<br/> `/ɹiːd/` (like 'reed'). You can get the second pronunciation in gruut<br/> with 'read_2' in your text (with word indexes enabled).<br/> <br/> > I would be interested to give that a go, is there anything I should<br/> > bear in mind when trying it?<br/> <br/> Thanks! Gruut has two stages: (1) tokenization and (2) phonemization.<br/> The command-line tool takes text in the first stage and produces JSON<br/> for the second. You can skip the first stage if you know exactly what<br/> you want:<br/> <br/> $ echo '{ 'clean_words': ['this', 'is', 'a', 'test'] }' | bin/gruut en-us phonemize | jq .pronunciation<br/> [<br/> [<br/> 'ð',<br/> 'ɪ',<br/> 's'<br/> ],<br/> [<br/> 'ɪ',<br/> 'z'<br/> ],<br/> [<br/> 'ə'<br/> ],<br/> [<br/> 't',<br/> 'ɛ',<br/> 's',<br/> 't'<br/> ]<br/> ]<br/> <br/> > Happy to move this discussion into a separate thread if that's better.<br/> <br/> Might be a good idea<br/> I'd be interested in feedback for non-English languages especially.<br/> <br/> [Gruut discussions<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [November 24, 2020, 6:00pm]<br/> <br/> I guess to accommodate your models first we need to enable Gruut in TTS.<br/> <br/> But maybe gruut and phonemizer generated the same outputs or at least<br/> they use the same IPA characters. In that case, we can replace Gruut<br/> with phonemizer to kick start your models in TTS.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> synesthesiam<br/> <DATE> [November 25, 2020, 3:18am]<br/> <br/> As I get better at training models too, I'd also be happy to some<br/> phonemizer-based models for the community.<br/> <br/> I have a 2080 Ti and 3x1060's (6GB). Any tips on how I might get models<br/> trained as fast as possible?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [November 25, 2020, 2:16pm]<br/> <br/> two tricks for training faster<br/> <br/> 1. Using gradual training<br/> 2. Using mixed_precision (as in dev branch) which would fit more<br/> instances in a batch.<br/> 3. Finetunning the model you trained before.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [December 8, 2020, 10:49pm]<br/> <br/> Two quick questions:<br/> <br/> 1. What is a reasonable way to host the model itself? Is something like<br/> a link to a shared Google Drive folder alright?<br/> <br/> 2. What are your thoughts on updating a model? I have a Tacotron2 model<br/> with an accompanying wavegrad vocoder - it's able to produce pretty<br/> good quality output ([like<br/> this](https://soundcloud.com/user-726556259/fractals-wavegrad-sample))<br/> but the current model has a few issues with the stopnet and<br/> sometimes with alignment, which I think I could improve upon. I'm<br/> wondering whether it's better to **a)** post sooner with the current<br/> one and then update it, or **b)** wait until I've got an improved<br/> version.<br/> <br/> I'm leaning towards a) but happy to go with b) if that's easier and it<br/> may avoid raising people's hopes only for them to run into cases where<br/> the model does poorly (which are reasonably common at the moment;<br/> examples of a few issues can be heard at times in [this<br/> sample](https://soundcloud.com/user-726556259/q-learning-wavegrad-sample);<br/> often small tweaks to the sentence can help but I'm hoping to get it<br/> beyond the need for that)<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [December 11, 2020, 1:42pm]<br/> <br/> I've got a TTS model and vocoder I'm happy to share here:<br/> <br/> **EK1:**<br/> <https://drive.google.com/drive/folders/1K8g9Lh23MBWSU2IqinrMEnWCT3ZlGV2W?usp=sharing><br/> <br/> Produced / usable with this commit<br/> [c802255](https://github.com/mozilla/TTS/commit/c80225544e2fb43abbccd94148cc2045d95f8f63) *slash* <br/> The dataset is from M-AILABS / LibriVox - the Queen's English set<br/> available<br/> [here](https://www.caito.de/2019/01/the-m-ailabs-speech-dataset/)<br/> <br/> The earlier issues I ran into seem largely absent in the latest r=2<br/> checkpoint and the overall quality is good at that stage when used with<br/> the vocoder. They'd crept in when transitioning to r=1.<br/> <br/> Here are two samples (one of which is the same text it was slightly<br/> struggling with in the earlier link I provided in my other comment)<br/> <br/> [Q-Learning<br/> example](https://soundcloud.com/user-726556259/q-learning-wavegrad-sample-260k)<br/> (*260k / 200 iterations in vocoder*) *slash* <br/> [Sherlock<br/> sample](https://soundcloud.com/user-726556259/sherlock-wavegrad-sample-260k)<br/> (*260k / 200 iterations in vocoder*)<br/> <br/> As usual you can speed it up (at the cost of quality) by bring the<br/> vocoder iterations down<br/> <br/> **NB:** To use it for inference with the server a small change is need -<br/> [see<br/> here](https://github.com/mozilla/TTS/issues/518#issuecomment-727647275)<br/> )<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> TheDayAfter<br/> <DATE> [December 13, 2020, 8:28am]<br/> <br/> Except for the background noise which turns on and off when speaking the<br/> examples sound great and especially natural, I like it. M-AILABS /<br/> LibriVox seems to contain some data jewels.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> synesthesiam<br/> <DATE> [December 16, 2020, 3:33am]<br/> <br/> , would you be OK with me<br/> including this model in [my docker<br/> images](https://github.com/synesthesiam/docker-mozillatts)?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [December 16, 2020, 8:08am]<br/> <br/> I would be<br/> delighted to see it included, but thank you for checking.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> synesthesiam<br/> <DATE> [December 16, 2020, 8:11pm]<br/> <br/> Did you upsample the WAV files from M-AI Labs? I noticed your model is<br/> 22050Hz, but the files (at least mine) are 16Khz.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [December 16, 2020, 8:59pm]<br/> <br/> Yes, sorry forgot to mention that!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> othiele<br/> <DATE> [December 17, 2020, 10:46am]<br/> <br/> Hm, facing the same challenge to upsample.<br/> what tool did you use for<br/> that? ffmpeg, sox, ...?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [December 17, 2020, 12:39pm]<br/> <br/> To convert from the 16kHz supplied to the 22.050kHz desired (to match<br/> other models/vocoders I had) I used sox, with this line:<br/> <br/> > for f in wavs/ *slash* *.wav; do sox ' *slash* $f' -r 22050 'wavs_22050/ *slash* ${f%%}'; done<br/> <br/> I think it puts the output in a wavs folder inside the wavs_22050<br/> folder, so you would need to create both before running it; if I'd had<br/> time I'd have looked at a way to avoid that (it's probably easy for any<br/> bash masters!<br/> <br/> The step before was simply combining all the speaker's wav files into a<br/> single folder from the two books (the naming convention means they don't<br/> collide). Then I merged the metadata CSV files and did the usual shuffle<br/> and split into training and test. In case anyone hasn't seen I included<br/> the training and test metadata files I used in with model in the Google<br/> Drive folder, which might save a little time.<br/> <br/> Lastly one basic change I *did* apply to the metadata files was to<br/> remove the chapter introduction lines ('Chapter One...'), because the<br/> narrator reads them more in her native accent than the main narration.<br/> It might be a bit too manual but I suspect a little more could done to<br/> improve things by removing samples where a clearly distinct accent is<br/> used (I have a feeling some characters are Irish or Scottish sounding) -<br/> however with something near to the 40hrs+ of audio it didn't seem to be<br/> too bad, although there are some alignment issues which I suspect are<br/> from pauses and may be contributed too by other factors like the accent<br/> issues (but that's just a guess!)<br/> <br/> Hope that helps!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> othiele<br/> <DATE> [December 17, 2020, 12:43pm]<br/> <br/> Thanks a lot, sox seems to be faster, but I thought I have read<br/> somewhere that it sometimes is not as 'good' as ffmpeg. Buth that could<br/> have been for videos. Thanks again, will try that for some of my<br/> material.<br/> <br/> [Archived Post]<br/>
